src/com/opc_ua/opcua_local_handler.cpp

```diff
@@ -155,22 +158,33 @@ void COPC_UA_Local_Handler::generateServerStrings(TForteUInt16 paUAServerPort, U
 
 }
 
-void COPC_UA_Local_Handler::configureUAServer(UA_ServerStrings &paServerStrings, UA_ServerConfig &paUaServerConfig) const {
-
-  paUaServerConfig.logger = COPC_UA_HandlerAbstract::getLogger();
-
+void COPC_UA_Local_Handler::configureUAServer(UA_ServerStrings &paServerStrings, UA_ServerConfig &paUaServerConfig, UA_UInt16 paServerPort) const {
 #ifdef FORTE_COM_OPC_UA_MULTICAST
   paUaServerConfig.applicationDescription.applicationType = UA_APPLICATIONTYPE_DISCOVERYSERVER;
   // hostname will be added by mdns library
   UA_String_clear(&paUaServerConfig.mdnsConfig.mdnsServerName);
   paUaServerConfig.mdnsConfig.mdnsServerName = UA_String_fromChars(paServerStrings.mMdnsServerName.c_str());
 #endif //FORTE_COM_OPC_UA_MULTICAST
 
-  UA_String_clear(&paUaServerConfig.customHostname);
+  UA_Array_delete(paUaServerConfig.serverUrls, paUaServerConfig.serverUrlsSize, &UA_TYPES[UA_TYPES_STRING]);
+  paUaServerConfig.serverUrls = nullptr;
+  paUaServerConfig.serverUrlsSize = 0;
+
+  UA_String serverUrls[1];
+  size_t serverUrlsSize = 0;
+  char serverUrlBuffer[512];
 #ifdef FORTE_COM_OPC_UA_CUSTOM_HOSTNAME
-  UA_String customHost = UA_String_fromChars(paServerStrings.mHostname.c_str());
-  UA_String_copy(&customHost, &paUaServerConfig.customHostname);
+  snprintf(serverUrlBuffer, sizeof(serverUrlBuffer), "opc.tcp://%s:%u", FORTE_COM_OPC_UA_CUSTOM_HOSTNAME, paServerPort);
```

Comment:
This changes the behavior of the OPC-UA server regarding how the hostname is determined in case a custom hostname is set in the CMake configuration; before a string consisting of _forte and the forte-communication-port was added (https://github.com/eclipse-4diac/4diac-forte/pull/260/files#diff-edd355cac746cc077b5f13abb9b7c46b6c05082bdd8aba0717d093a2ac9fca34R150-R157)

As this behavior was seen as confusing before I am not against this change; I only want to make sure that this does not break things in some application

Reply:
@cochicde do you have any input on this topic as the earliest git commits on hostname generation trace this code to you?

Reply:
I can't remember why it was set like `forte_{PORT}` and I would assume it wasn't planned, it was just some decision based on nothing in particular. 

In this case, however, I would keep it consistent and:
1. Keep the string generation inside the `generateServerStrings`
2. Don't change things related to the hostname generation in this PR, but focus only on the update to the new stack version. I'd move the hostname change to a new PR for simplicity

Reply:
I'd still keep the all the string handling of the names under the `generateServerStrings` otherwise you two different part of the code doing the same thing, but differently

Reply:
We should probably keep the old format `customName-forte_{PORT}` then.

Reply:
I moved the generation of the hostname to the `generateServerStrings` method

Reply:
LGTM